test/gtest/ucp/test_ucp_am.cc

```diff
@@ -820,6 +820,210 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
+        request_wait(pending_sptr);
+        wait_for_value(&cb_counter, req_counter);
+        request_wait(rndv_pending_sptr);
+        EXPECT_EQ(req_counter, cb_counter);
+
+        if (prereg()) {
+            sender().mem_unmap(memh);
+        }
+    }
+
+    static ucs_status_t
+    am_data_cb(void *arg, const void *header, size_t header_length, void *data,
+               size_t length, const ucp_am_recv_param_t *param)
+    {
+        test_ucp_am_nbx_send_copy_header *self =
+                static_cast<test_ucp_am_nbx_send_copy_header*>(arg);
+        return self->am_data_handler(header, header_length, data, length,
+                                     param);
+    }
+
+    /// @override
+    ucs_status_t
+    am_data_handler(const void *header, size_t header_length, void *data,
+                    size_t length, const ucp_am_recv_param_t *rx_param)
+    {
+        ucs_status_t status;
+        m_am_received = false;
+        status = test_ucp_am_nbx::am_data_handler(header, header_length, data,
+                                                  length, rx_param);
+        cb_counter++;
+        return status;
+    }
+
+    virtual unsigned get_send_flag() const
+    {
+        if (get_variant_value(2) & UCP_AM_SEND_FLAG_REPLY) {
+            return get_variant_value(2);
+        }
+
+        return 0;
+    }
+
+private:
+    /// @override
+    ucs_status_ptr_t send_am(const ucp::data_type_desc_t &dt_desc,
+                             unsigned flags = 0, const void *hdr = NULL,
+                             unsigned hdr_length = 0,
+                             const ucp_mem_h memh = NULL)
+    {
+        req_counter++;
+        return test_ucp_am_nbx::send_am(dt_desc, flags, hdr, hdr_length, memh);
+    }
+
+    ucs_status_ptr_t fill_sq(ucp::data_type_desc_t &sdt_desc,
+                             const ucp_mem_h memh = NULL,
+                             size_t header_size = 0ul, unsigned flags = 0)
+    {
+        const auto timeout = 5;
+        ucs_status_ptr_t pending_sptr;
+
+        // Warmup for wireup connection
+        pending_sptr = send_am(sdt_desc, get_send_flag() | flags,
+                               m_hdr_copy.data(), header_size, memh);
+        wait_for_flag(&m_am_received);
+        request_wait(pending_sptr);
+        pending_sptr = NULL;
+
+        const ucs_time_t deadline = ucs::get_deadline(timeout);
+        while ((ucs_get_time() < deadline) && (pending_sptr == NULL)) {
+            pending_sptr = send_am(sdt_desc, get_send_flag() | flags,
+                                   m_hdr_copy.data(), header_size, memh);
+        }
+
+        return pending_sptr;
+    }
+
+    size_t req_counter = 0;
+    size_t cb_counter  = 0;
+    std::string m_hdr_copy;
+};
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, send_short, "ZCOPY_THRESH=-1",
+           "RNDV_THRESH=-1")
+{
+    check_eager(8, 8);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, bcopy_single, "ZCOPY_THRESH=-1",
```

Comment:
can we unite tests with similar configuration settings to reduce testing time?
see https://github.com/openucx/ucx/pull/8341